server/text_generation_server/models/__init__.py

```diff
@@ -702,7 +707,21 @@ def get_model(
 
     if model_type == MISTRAL:
         sliding_window = config_dict.get("sliding_window", -1)
-        if FLASH_ATTENTION:
+        if (
+            (sliding_window is not None and sliding_window != -1)
+            and not SUPPORTS_WINDOWING
+            and max_batch_prefill_tokens > sliding_window
+        ):
+            raise ValueError(
+                SLIDING_WINDOW_MESSAGE.format(
```

Comment:
```suggestion
                f"The backend {SYSTEM} does not support sliding window attention. TGI webserver was started max_batch_prefill_tokens={max_batch_prefill_tokens} larger than sliding_window={sliding_window}. To use this model with the {SYSTEM} backend, please launch TGI with the argument `--max-batch-prefill-tokens` smaller than {sliding_window}."
```

Reply:
I think we can factor this out to have a single location for the error message.


We get sliding window *before* the massive switch.
Then we see if we need windowing.

```python
sliding_window = config_dict.get("sliding_window", -1)
needs_windowing = (sliding_window is not None and sliding_window != -1) and max_input_tokens > sliding_window
if needs_windowing and not SUPPORTS_WINDOWING:
    FLASH_ATTENTION = False
    # or raise 
   'The backend {SYSTEM} ...
```

Wouldn't that simplify massively things? 

Also I wasn't aware those other models also used windowing, I thought only mistral did (not even mixtral)